src/main/java/bio/terra/app/logging/LoggerInterceptor.java

```diff
@@ -14,40 +16,75 @@
 
 @Component
 public class LoggerInterceptor implements HandlerInterceptor {
-  private static Logger logger = LoggerFactory.getLogger(LoggerInterceptor.class);
+  private static final Logger logger = LoggerFactory.getLogger(LoggerInterceptor.class);
+  // Constants for requests coming in that aren't authenticated
+  private static final String UNAUTHED_USER_ID = "N/A";
+  private static final String UNAUTHED_EMAIL = "N/A";
+  private static final String UNAUTHED_INSTITUTE = "N/A";
+  // Don't log requests for URLs that end with any of   the following paths
+  private static final Set<String> LOG_EXCLUDE_LIST = Set.of("/status");
+
+  private static final String REQUEST_START_ATTRIBUTE = "x-request-start";
+  private static final Long NOT_FOUND_DURATION = -1L;
+
   private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
 
   @Autowired
   public LoggerInterceptor(AuthenticatedUserRequestFactory authenticatedUserRequestFactory) {
     this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
   }
 
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+      throws Exception {
+    request.setAttribute(REQUEST_START_ATTRIBUTE, System.currentTimeMillis());
+    return true;
+  }
+
   @Override
   public void afterCompletion(
       HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex)
       throws Exception {
-    AuthenticatedUserRequest userReq = authenticatedUserRequestFactory.from(request);
-    String userId = userReq.getSubjectId();
-    String userEmail = userReq.getEmail();
-    String institute = userEmail != null ? userEmail.substring(userEmail.indexOf("@") + 1) : null;
+
+    String userId;
+    String userEmail;
+    String institute;
+    try {
+      AuthenticatedUserRequest userReq = authenticatedUserRequestFactory.from(request);
+      userId = userReq.getSubjectId();
+      userEmail = userReq.getEmail();
+      institute = userEmail != null ? userEmail.substring(userEmail.indexOf("@") + 1) : null;
```

Comment:
It seems odd to me that we're logging both the `institute` and the email, since the email contains that value.

Reply:
Yeah...I left that in place but I agree...I don't think it gives us much.  If we want to track it, we should probably set a sink